src/pybind/mgr/nfs/cluster.py

```diff
@@ -139,10 +139,12 @@ def delete_nfs_cluster(self, cluster_id: str) -> Tuple[int, str, str]:
                 completion = self.mgr.remove_service('nfs.' + cluster_id)
                 orchestrator.raise_if_exception(completion)
                 self.delete_config_obj(cluster_id)
-                return 0, "", ""
-            return 0, "", "Cluster does not exist"
+                return
+            raise ErrorResponse("Cluster does not exist",
```

Comment:
Hmm. That was probably an oversight, but I'm not sure what's best, retaining the somewhat strange old behavior of sending the success error code when this is clearly an error or trying to avoid any behavior change here. Would be happy to hear your opinion.

Reply:
I guess it depends if we think it's more important for the command to be idempotent or to make sure users are aware that the cluster they deleted isn't there anymore. We actually had a command like this in cephadm that would just print success if you tried to delete a nonexistent service, but eventually somebody requested we have it print an error when you do that. So I guess maybe raising here is the right thing.

Reply:
Yeah, my subconscious was probably encouraging the idea that errors should be errors. Otherwise it is not worth reporting at all. :-)

Reply:
@ajarr flagged this in our standup today. See my review comment. :)